System/PathFinder.h

```diff
@@ -127,7 +127,7 @@ namespace RTE {
 		/// <param name="digStrength">What material strength the search is capable of digging through.</param>
 		/// <param name="callback">The callback function to be run when the path calculation is completed.</param>
 		/// <returns>A shared pointer to the volatile PathRequest to be used to track whehter the asynchrnous path calculation has been completed, and check its results.</returns>
-		std::shared_ptr<volatile PathRequest> CalculatePathAsync(Vector start, Vector end, float digStrength, PathCompleteCallback callback = nullptr);
+		std::shared_ptr<volatile PathRequest> CalculatePathAsync(Vector start, Vector end, float digStrength, const PathCompleteCallback &callback = nullptr);
```

Comment:
Passing a const reference that is default-initialized is insane and I'm gobsmacked that this compiles. If we're doing this, use a separate overloaded function to pass in a callback to make the lifetime of the object clear.